### PR TITLE
Fix Issue 23776 - getSymbolsByUDA fails to fetch symbols from module

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

File: std/traits.d [hunk 1]
```diff
@@ -8812,6 +8812,20 @@ template getSymbolsByUDA(alias symbol, alias attribute)
...
+{
+    struct T
+    {
+        struct Tag {}
+        @Tag struct MyStructA {}
+        @Tag struct MyStructB {}
+        @Tag struct MyStructC {}
+    }
+    alias components = getSymbolsByUDA!(T, T.Tag);
+    static assert(components.length > 0);
```
Comment (maxhaton) [inline std/traits.d:8826]:
```suggestion
    static assert(components.length > 0);
```

**Review by maxhaton — CHANGES_REQUESTED** [review]

File: std/traits.d [hunk 2]
```diff
@@ -8812,6 +8812,20 @@ template getSymbolsByUDA(alias symbol, alias attribute)
     static assert(is(getSymbolsByUDA!(X, X) == AliasSeq!()));
 }
 
+// https://issues.dlang.org/show_bug.cgi?id=23776
+@safe pure nothrow @nogc unittest
+{
+    struct T
```
[the remark this answers is not in the document]
Comment (RazvanN7) [reply std/traits.d:8818]: It doesn't matter if the symbols are in a different module or not. The previous version of the condition simply discarded type declarations.

File: std/traits.d (hunk 1 above)
Comment (RazvanN7) [reply std/traits.d]: don't shoot, not my fault. @quickfur had a trailing white space in his bug report, I just pasted it. :-"

File: std/traits.d [hunk 3]
```diff
@@ -8866,7 +8880,7 @@ private template getSymbolsByUDAImpl(alias symbol, alias attribute, names...)
             alias member = __traits(getMember, symbol, names[0]);
 
             // Filtering not compiled members such as alias of basic types.
-            static if (isAliasSeq!member || isType!member)
+            static if (isAliasSeq!member || (isType!member && !isAggregateType!member))
```
Comment (pbackus) [inline std/traits.d:8883]: [User-defined types][1] include both aggregates and enums:

```suggestion
            static if (isAliasSeq!member || (isType!member && !isAggregateType!member && !is(member == enum)))
```

[1]: https://dlang.org/spec/type.html#user-defined-types

**Review by pbackus — CHANGES_REQUESTED** [review]
Even with this change it still fails to fetch `enum`s with UDAs. Here's the failing test case:

```d
@safe pure nothrow @nogc unittest
{
    struct T
    {
        struct Tag {}
        @Tag enum MyEnumA;
        @Tag enum MyEnumB;
        @Tag enum MyEnumC;
    }
    alias components = getSymbolsByUDA!(T, T.Tag);
    static assert(components.length > 0);
}
```

**Comment by RazvanN7** [comment]
@pbackus Thanks! Should be good now.

**Review by pbackus — APPROVED** [review]

File: std/traits.d (hunk 2 above)
[the remark this answers is not in the document]
Comment (ibuclaw) [reply std/traits.d:8818]: Compiling this unittest wthout patch
```
std/traits.d(8826): Error: static assert:  `0LU > 0LU` is false
```
With patch the unittest compiles successfully.

**Pull request merged**